### PR TITLE
fix: MLS degraded dialogs [WPB-6607]

### PR DESCRIPTION
**Pull request by borichellow**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6607" title="WPB-6607" target="_blank">BugWPB-6607</a>  [Android] Show alert message after conversation becomes degraded
  </td></table>
  <br />
 

# What's new in this PR?

### Issues

When a conversation becomes degraded, we must show a message to the user when:

Expected:

they try to send a message
they try to start a call
stop the ongoing call
Actual: None of the above happen!

### Causes (Optional)

Was not implemented on kalium side + this dialog should be shown every time user sends a message AND didn't click "Send anywhere" yet

### Solutions

Update kalium + improve dialog dismiss logic

### Waiting for update kalium 
https://github.com/wireapp/kalium/pull/2501

### CODE REVIEW

**Review by alexandreferris — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [3156](https://10.10.124.134/job/AR-build-pipeline/3156/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.0(8904230)(8904230)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.0(8904230)(8904230)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageComposerViewModel.kt [hunk 1]
```diff
@@ -527,7 +527,10 @@ class MessageComposerViewModel @Inject constructor(
     fun dismissSureAboutSendingMessage() {
         (sureAboutMessagingDialogState as? SureAboutMessagingDialogState.Visible)?.let {
             viewModelScope.launch {
-                it.markAsNotified()
+                if (it is SureAboutMessagingDialogState.Visible.ConversationUnderLegalHold) {
+                    setNotifiedAboutConversationUnderLegalHold(conversationId)
+                }
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageComposerViewModel.kt]: suggestion: maybe just for the consistency we should also have the same logic for the dialog informing about conversation being under legal hold from now - so just these 3 lines to be removed 🤔

Comment (borichellow) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageComposerViewModel.kt]: good point :)

**Review by saleniuk — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [3162](https://10.10.124.134/job/AR-build-pipeline/3162/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.0(8904820)(8904820)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.0(8904820)(8904820)-staging-compat.apk)

**Pull request merged**